### ISSUE TITLE
Add @var <variable_name> does not show up in Magicento menu

### ISSUE THREAD

**Issue by kidonchu** [enriquepiatti/Magicento#110]
I am using Windows 8.1, PhpStorm 7.1.3, Magicento 2.6.5.
I have following code in my model function.
$collection = Mage::getModel('catalog/product')->getCollection();
Having my cursor on $collection, if I press Alt+M, I don't see the option "Add @var $collection".
What could cause this issue? I only see the options shown in the image.
![issue](https://f.cloud.github.com/assets/1875398/2279455/7ab9a7a8-9f6f-11e3-9af3-c3248022bbfa.PNG)

**Comment by enriquepiatti** [comment]
Hi @kidonchu, I think this is related with some changes in PhpStorm, anyway, I was thinking in removing that feature, is not so useful anymore because Magicento can detect the type of $collection automatically now, and it can cause some problems, because the @var docs have priority over the types inferred by Magicento, so for example if you change the code to something like:  
$collection = Mage::getModel('catalog/category')->getCollection(); 
but you keep the @var doc saying it's a product collection, then even when Magicento understands it's a category collection now, PhpStorm will override it and interpret it as a product collection.     

Are you needing that @var doc for some reason? is not Magicento getting the right type automatically?

**Comment by kidonchu** [comment]
The only reason that I would need @var doc is documentation. Other than that, Magicento is just great. If it's due to some changes with PhpStorm , I guess I just need to use PhpStorm's built-in Alt+Enter feature to insert @var doc. 
I agree with the opinion that @var overriding Magicento logic is not desired behavior. Thanks for the great plugin.

**Comment by enriquepiatti** [comment]
No, the original ALT+ENTER is not working automatically, at least from my tests, I will try to add an extension for that intention on the next release.

**Comment by enriquepiatti** [comment]
Added a new intention for this in v2.7, it should work better than the one from PhpStorm

**Comment by ihor-sviziev** [comment]
@enriquepiatti it's working not in all cases. Example:
Working:

``` php
$regionModel = Mage::getModel('directory/region');
```

Not working:

``` php
$regionModel = Mage::getModel('directory/region')->loadByCode($state, $countryCode);
```

In second case I have message 

```
Magicento Notification: Add @var Intention: Cannot find ClassName
```

**Comment by enriquepiatti** [comment]
Hi @igor-svizev yes you are right, Magicento won't be able to find the correct class right now in a lot of situations, usually that situation is when you are chaining another method like in your example. Magicento can detect some common chains like "load" or "getCollection" or "getResource" etc, but it doesn't have support for every chained method, I'm working on that but I need to find a way to make that efficiently.
